### ISSUE TITLE
Zammad UI is completely stuck after selecting overview "All tickets"

### ISSUE THREAD

**Issue by MartinBorg** [zammad/zammad#989]
Zammd: 1.5.0-1493106761.5ac346e.jessie (deb)
OS: Debian Jessie amd64
Browser: Firefox 45.8.0esr or Chromium 57.0.2987.98 on Debian Jessie amd64

### Expected behaviour:

* click on overview "All tickets" shows all tickets (currently ~16 tickets).

### Actual behaviour:

* click on overview "All tickets" leads to empty overview, i.e. there is not even the table itself, less so any tickets. Any clicks on other elements, like "Dashboard" show no reaction anymore. Only hard reload of the page makes it work again, until one selects an overview again.

### Steps to reproduce the behaviour:

* See above. Behaviour is 100% reproducible so far. Note, that I did not see this behaviour before, in version 1.4.

**Comment by martini** [comment]
Hi @MartinBorg 

thanks for this issue. Can you tell me, is there something in the console error log of the browser?

 -Martin

**Comment by MartinBorg** [comment]
Oh, yes, quite a lot:

```
|  _____                                    _
| / _  / __ _ _ __ ___  _ __ ___   __ _  __| |
| \// / / _` | '_ ` _ \| '_ ` _ \ / _` |/ _` |
|  / //\ (_| | | | | | | | | | | | (_| | (_| |
| /____/\__,_|_| |_| |_|_| |_| |_|\__,_|\__,_|
|
| Hi there, nice to meet you!
|
| Visit "https://zammad.org/participate" and let's make Zammad better.
|
| The Zammad Team.
| application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:21:19077
App.n(notice)                  | view: my_assigned s application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:21:19077
App.n(notice)                  | view: all_open s application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:21:19077
App.n(notice)                  | view: all_tickets s application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:21:19077
ERROR, cant find customer_id App.User.find(4) for 'Ticket' Test with picture application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:21:19077
App.n(error)                   | route ticket/view/:view: TypeError: t is null
Stack trace:
t</n.viewPrintItem@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:21:7723
t</n.viewPrint@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:21:7387
t</s.P@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:21:4529
this.JST["app/views/generic/table_row"]/</<@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:49:17054
this.JST["app/views/generic/table_row"]/<@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:49:15893
this.JST["app/views/generic/table_row"]@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:49:15881
t</n.view/e@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:21:8696
App.ControllerTable</i.prototype.tableGen@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:25:24218
t/<@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:25:19747
App.ControllerTable</i.prototype.render@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:25:21269
t/<@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:25:19747
i@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:25:20836
n</n.prototype.render@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:38:16528
s/<@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:37:24286
t</t.prototype.callback@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:60:23571
e/<@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:60:22004
t</t.prototype.bind@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:60:22719
App.OverviewListCollection</e.bind@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:60:23816
n</n.prototype.update@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:38:14001
s/<@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:37:24286
App.TicketOverview</i.prototype.show@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:38:7856
s/<@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:37:24286
t</i.prototype.show@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:61:5223
e/<@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:60:31467
t</i.prototype.showControllerHideOthers@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:61:4713
e/<@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:60:31467
t</i.prototype.startController@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:61:4425
e/<@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:60:31467
t</i.prototype.executeSingel@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:61:4016
t</i.prototype.run@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:61:2681
t</i.prototype.execute@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:61:2539
App.TaskManager</e.execute@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:60:31947
e@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:38:26300
n/i</</</i</<@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:60:18935
n/i</</<@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:60:18875
r</t.prototype.proxy/</<@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:7:28709
e</e.prototype.match@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:8:16014
n</s.prototype.matchRoute@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:8:18629
n</s.matchRoutes@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:8:18402
n</s.change@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:8:18257
rt.event.dispatch@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:2:18724
rt.event.add/m.handle@https://zammad.ammonit.com/assets/application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:2:16834
 application-7c3a37c46fd5e2e433a72e92e09db23360dd1fcf8a584130ad65976a129d8e63.js:21:19077
```

**Comment by MartinBorg** [comment]
Ah, I see, there is a user (id=4) missing, right? I might have deleted it!

**Comment by martini** [comment]
> Ah, I see, there is a user (id=4) missing, right? I might have deleted it!

Yes. The customer of the ticket is missing in the web frontend. So the web frontend can't rendered (because of missing reference data).

Can you execute the following:

`zammad run rails r "p User.find(4)"`

**Comment by MartinBorg** [comment]
It does not exist anymore. I `destroyed` it. Can I find the ticket somehow and destory it, too? It's a testing instance of zammad anyway, so there is nothing lost, if I just destroy arbitrary users or tickets :~)

**Comment by martini** [comment]
Delete all tickets:

`zammad run rails r "Ticket.all.destroy_all"`

Delete a certain ticket:

`zammad run rails r "Ticket.find(4).destroy"`

Update the customer of a bundle of tickets:

`zammad run rails r "Ticket.where(customer_id: 4).update_all(customer_id: 1)"`

PS: If you also deleted user with id 1, you should recreate your database (if it's a test instance). We also work on a feature to merge users and to destroy users without references (e. g. senders of spam mail which are already deleted).

**Comment by MartinBorg** [comment]
OK, I found the "bad" ticket, destroyed it, and everything works fine again. **Many thanks!**

IMHO, it would be nice, if the Zammad UI would behave more "robust" in such a situation, e.g. just not show the broken ticket, but still show all others or something. Maybe make this issue #989 a "request for improvement", not a bug?

Btw. I did not delete User id=1, but what is that user id about? Some "magic"?

**Comment by martini** [comment]
> IMHO, it would be nice, if the Zammad UI would behave more "robust" in such a situation, e.g. just not show the broken ticket, but still show all others or something. Maybe make this issue #989 a "request for improvement", not a bug?

You can call it "request for improvement" or "bug". Both need to be fixed/implemented. :)

> Btw. I did not delete User id=1, but what is that user id about? Some "magic"?

It's the first created user. -> You should not delete database records manually.

**Comment by sirluis** [comment]
> Delete all tickets:
> zammad run rails r "Ticket.all.destroy_all"
> 
> Delete a certain ticket:
> zammad run rails r "Ticket.find(4).destroy"
> 
> Update the customer of a bundle of tickets:
> zammad run rails r "Ticket.where(customer_id: 4).update_all(customer_id: 1)"

Hi! Can I run this commands to remove spam tickets/users (remove all tickets from a user is owner, then remove the account) in production?

**Comment by martini** [comment]
Behaviour will be improved/fixed in upcoming Zammad 2.1. Thanks!
